lightpath/tests/conftest.py

```diff
@@ -306,7 +290,19 @@ def lcls():
 def lcls_client():
     db = os.path.join(os.path.dirname(__file__), 'path.json')
     print(db)
-    return happi.Client(path=db)
+    client = happi.Client(path=db)
+
+    # monkeypatch to never use cache to get device
+    # cached devices will retain callback information between tests
+    old_get = happi.SearchResult.get
```

Comment:
Would recommend using the `monkeypatch` pytest fixture; it's pretty nice and handles the cleanup for you

`monkeypatch.setattr(happi.SearchResult, "get", new_get)`

(oh, I see you used it elsewhere... Well, the recommendation stands regardless)

Reply:
I don't know why I didn't do this..